src/plugins/data/common/search/es_search/types.ts

```diff
@@ -51,8 +52,37 @@ export interface IEsSearchResponse<Source = any> extends IKibanaSearchResponse {
    * Indicates whether the results returned are complete or partial
    */
   isPartial?: boolean;
-  rawResponse: SearchResponse<Source>;
+  rawResponse: SearchResponse<T>;
+  rawEqlResponse?: EqlSearchResponse<T>;
 }
 
 export const isEsResponse = (response: any): response is IEsSearchResponse =>
   response && response.rawResponse;
+
+export interface EqlSearchResponse<T> {
```

Comment:
I don't believe that this type exists in the `elasticsearch` package, or at least I couldn't find it: the EQL client is currently typed to return an [`ApiResponse`](https://github.com/elastic/elasticsearch-js/blob/master/api/kibana.d.ts#L182) which makes no assumptions about the response body.